prometheus/client.go

```diff
@@ -31,23 +31,27 @@ func NewClient() (*Client, error) {
 	if err != nil {
 		return nil, err
 	}
-	client.p8s = p8s
+	client.api = v1.NewAPI(p8s)
 	return &client, nil
 }
 
+// For testing / mocking
+func (in *Client) inject(api v1.API) {
+	in.api = api
+}
+
 // GetSourceServices returns a map of source services for a given service identified by its namespace and service name.
 // Returned map has a destination version as a key and a "<origin service>/<origin version>" pair as value.
 // Destination service is not included in the map as it is passed as argument.
 // It returns an error on any problem.
-func (in *Client) GetSourceServices(namespace string, servicename string) (map[string]string, error) {
+func (in *Client) GetSourceServices(namespace string, servicename string) (map[string][]string, error) {
```

Comment:
@lucasponce I did some changes in the prometheus client, can you please review:

- GetSourceServices now returns a "multimap", since there can be more than 1 source service per version.
- The client now creates the API object from the start and keeps it in memory (thus it doesn't need to store the internal `client` field). This is the way I found to be able to inject and mock API from tests. Maybe there are other ways to do, I don't know.